Subject: [PATCH] выполнил задание по созданию запросов на вещи
---
Index: src/main/java/ru/practicum/shareit/booking/dto/NewBookingRequest.java
===================================================================
diff --git a/src/main/java/ru/practicum/shareit/booking/dto/NewBookingRequest.java b/src/main/java/ru/practicum/shareit/booking/dto/NewBookingRequest.java
deleted file mode 100644
--- a/src/main/java/ru/practicum/shareit/booking/dto/NewBookingRequest.java	(revision 67e65e866b978068413d04956e35ebed1b747536)
+++ /dev/null	(revision 67e65e866b978068413d04956e35ebed1b747536)
@@ -1,26 +0,0 @@
-package ru.practicum.shareit.booking.dto;
-
-import jakarta.validation.constraints.Future;
-import jakarta.validation.constraints.FutureOrPresent;
-import jakarta.validation.constraints.NotNull;
-import jakarta.validation.constraints.Positive;
-import lombok.Data;
-import ru.practicum.shareit.util.StartBeforeEndDateValid;
-
-import java.time.LocalDateTime;
-
-@Data
-@StartBeforeEndDateValid(message = "Start date must be before end date")
-public class NewBookingRequest {
-
-    private Long id;
-    @NotNull
-    @FutureOrPresent(message = "Дата начала должна быть либо в настоящем времени, либо в будущем")
-    private LocalDateTime start;
-    @NotNull
-    @Future(message = "Дата окончания должна быть в будущем времени")
-    private LocalDateTime end;
-    @Positive
-    @NotNull
-    private Long itemId;
-}
Index: src/main/java/ru/practicum/shareit/item/ItemController.java
===================================================================
diff --git a/src/main/java/ru/practicum/shareit/item/ItemController.java b/src/main/java/ru/practicum/shareit/item/ItemController.java
deleted file mode 100644
--- a/src/main/java/ru/practicum/shareit/item/ItemController.java	(revision 67e65e866b978068413d04956e35ebed1b747536)
+++ /dev/null	(revision 67e65e866b978068413d04956e35ebed1b747536)
@@ -1,67 +0,0 @@
-package ru.practicum.shareit.item;
-
-import jakarta.validation.Valid;
-import lombok.RequiredArgsConstructor;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PatchMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestHeader;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.RestController;
-import ru.practicum.shareit.item.dto.CommentDto;
-import ru.practicum.shareit.item.dto.ItemDto;
-import ru.practicum.shareit.item.dto.ItemOwnerDto;
-import ru.practicum.shareit.item.dto.NewCommentRequestDto;
-import ru.practicum.shareit.item.dto.NewItemRequestDto;
-import ru.practicum.shareit.item.dto.UpdateItemRequestDto;
-
-import java.util.List;
-
-import static ru.practicum.shareit.util.HeaderConstant.USER_ID_HEADER;
-
-@RestController
-@RequestMapping("/items")
-@RequiredArgsConstructor
-public class ItemController {
-
-    private final ItemService itemService;
-
-    @PostMapping
-    public ItemDto createItem(@Valid @RequestBody NewItemRequestDto requestDto,
-                              @RequestHeader(USER_ID_HEADER) long userId) {
-        return itemService.createItem(requestDto, userId);
-    }
-
-    @PostMapping("/{itemId}/comment")
-    public CommentDto createComment(@Valid @RequestBody NewCommentRequestDto request,
-                                    @PathVariable long itemId,
-                                    @RequestHeader(USER_ID_HEADER) long userId) {
-        return itemService.createComment(request, itemId, userId);
-    }
-
-    @PatchMapping("/{itemId}")
-    public ItemDto updateItem(@Valid @RequestBody UpdateItemRequestDto requestDto,
-                              @PathVariable long itemId,
-                              @RequestHeader(USER_ID_HEADER) long itemOwnerId) {
-        return itemService.updateItem(requestDto, itemOwnerId, itemId);
-    }
-
-    @GetMapping("/{itemId}")
-    public ItemOwnerDto getItem(@PathVariable long itemId,
-                                @RequestHeader(USER_ID_HEADER) long ownerItemId) {
-        return itemService.getItem(itemId, ownerItemId);
-    }
-
-    @GetMapping
-    public List<ItemOwnerDto> getOwnersItems(@RequestHeader(USER_ID_HEADER) long userId) {
-        return itemService.getOwnersItems(userId);
-    }
-
-    @GetMapping("/search")
-    public List<ItemDto> searchItemsByText(@RequestParam(name = "text") String text) {
-        return itemService.searchItemsByText(text);
-    }
-}
Index: src/main/java/ru/practicum/shareit/util/CheckDateValidator.java
===================================================================
diff --git a/src/main/java/ru/practicum/shareit/util/CheckDateValidator.java b/src/main/java/ru/practicum/shareit/util/CheckDateValidator.java
deleted file mode 100644
--- a/src/main/java/ru/practicum/shareit/util/CheckDateValidator.java	(revision 67e65e866b978068413d04956e35ebed1b747536)
+++ /dev/null	(revision 67e65e866b978068413d04956e35ebed1b747536)
@@ -1,24 +0,0 @@
-package ru.practicum.shareit.util;
-
-import jakarta.validation.ConstraintValidator;
-import jakarta.validation.ConstraintValidatorContext;
-import ru.practicum.shareit.booking.dto.NewBookingRequest;
-
-import java.time.LocalDateTime;
-
-public class CheckDateValidator implements ConstraintValidator<StartBeforeEndDateValid, NewBookingRequest> {
-
-    @Override
-    public void initialize(StartBeforeEndDateValid constraintAnnotation) {
-    }
-
-    @Override
-    public boolean isValid(NewBookingRequest bookingRequest, ConstraintValidatorContext constraintValidatorContext) {
-        LocalDateTime start = bookingRequest.getStart();
-        LocalDateTime end = bookingRequest.getEnd();
-        if (start == null || end == null) {
-            return false;
-        }
-        return start.isBefore(end);
-    }
-}
\ No newline at end of file
Index: src/main/java/ru/practicum/shareit/util/StartBeforeEndDateValid.java
===================================================================
diff --git a/src/main/java/ru/practicum/shareit/util/StartBeforeEndDateValid.java b/src/main/java/ru/practicum/shareit/util/StartBeforeEndDateValid.java
deleted file mode 100644
--- a/src/main/java/ru/practicum/shareit/util/StartBeforeEndDateValid.java	(revision 67e65e866b978068413d04956e35ebed1b747536)
+++ /dev/null	(revision 67e65e866b978068413d04956e35ebed1b747536)
@@ -1,24 +0,0 @@
-package ru.practicum.shareit.util;
-
-
-import jakarta.validation.Constraint;
-import jakarta.validation.Payload;
-
-import java.lang.annotation.Documented;
-import java.lang.annotation.ElementType;
-import java.lang.annotation.Retention;
-import java.lang.annotation.Target;
-
-import static java.lang.annotation.RetentionPolicy.RUNTIME;
-
-@Target(ElementType.TYPE)
-@Retention(RUNTIME)
-@Documented
-@Constraint(validatedBy = {CheckDateValidator.class})
-public @interface StartBeforeEndDateValid {
-    String message() default "Start must be before end or not null";
-
-    Class<?>[] groups() default {};
-
-    Class<? extends Payload>[] payload() default {};
-}
\ No newline at end of file
Index: pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/pom.xml b/pom.xml
--- a/pom.xml	(revision 67e65e866b978068413d04956e35ebed1b747536)
+++ b/pom.xml	(date 1728366690302)
@@ -11,6 +11,7 @@
 
 	<groupId>ru.practicum</groupId>
 	<artifactId>shareit</artifactId>
+	<packaging>pom</packaging>
 	<version>0.0.1-SNAPSHOT</version>
 
 	<name>ShareIt</name>
@@ -19,114 +20,29 @@
 		<java.version>21</java.version>
 	</properties>
 
-	<dependencies>
-		<dependency>
-			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-web</artifactId>
-		</dependency>
-		<dependency>
-			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-actuator</artifactId>
-		</dependency>
-		<dependency>
-			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-configuration-processor</artifactId>
-			<optional>true</optional>
-		</dependency>
-		<dependency>
-			<groupId>org.postgresql</groupId>
-			<artifactId>postgresql</artifactId>
-			<scope>runtime</scope>
-		</dependency>
-		<dependency>
-			<groupId>com.h2database</groupId>
-			<artifactId>h2</artifactId>
-			<scope>test</scope>
-		</dependency>
-		<dependency>
-			<groupId>com.querydsl</groupId>
-			<artifactId>querydsl-jpa</artifactId>
-			<classifier>jakarta</classifier>
-			<version>5.1.0</version>
-		</dependency>
-		<dependency>
-			<groupId>org.projectlombok</groupId>
-			<artifactId>lombok</artifactId>
-			<optional>true</optional>
-		</dependency>
-		<dependency>
-			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-test</artifactId>
-			<scope>test</scope>
-		</dependency>
-		<dependency>
-			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-validation</artifactId>
-		</dependency>
-		<dependency>
-			<groupId>org.mapstruct</groupId>
-			<artifactId>mapstruct</artifactId>
-			<version>1.6.0.Beta1</version>
-		</dependency>
-		<dependency>
-			<groupId>org.mapstruct</groupId>
-			<artifactId>mapstruct-processor</artifactId>
-			<version>1.6.0.Beta1</version>
-			<scope>provided</scope>
-		</dependency>
-		<dependency>
-			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-data-jpa</artifactId>
-		</dependency>
-	</dependencies>
+	<modules>
+		<module>gateway</module>
+		<module>server</module>
+	</modules>
 
 	<build>
-		<resources>
-			<resource>
-				<directory>src/main/resources</directory>
-				<filtering>true</filtering>
-			</resource>
-		</resources>
-		<plugins>
-			<plugin>
-				<groupId>com.mysema.maven</groupId>
-				<artifactId>apt-maven-plugin</artifactId>
-				<version>1.1.3</version>
-				<executions>
-					<execution>
-						<goals>
-							<goal>process</goal>
-						</goals>
-						<configuration>
-							<outputDirectory>target/generated-sources/java</outputDirectory>
-							<processor>com.querydsl.apt.jpa.JPAAnnotationProcessor</processor>
-						</configuration>
-					</execution>
-				</executions>
-				<dependencies>
-					<dependency>
-						<groupId>com.querydsl</groupId>
-						<artifactId>querydsl-apt</artifactId>
-						<classifier>jakarta</classifier>
-						<version>5.1.0</version>
-					</dependency>
-				</dependencies>
-			</plugin>
-			<plugin>
-				<groupId>org.springframework.boot</groupId>
-				<artifactId>spring-boot-maven-plugin</artifactId>
-				<configuration>
-					<excludes>
-						<exclude>
-							<groupId>org.projectlombok</groupId>
-							<artifactId>lombok</artifactId>
-						</exclude>
-					</excludes>
-				</configuration>
-			</plugin>
-		</plugins>
-		<pluginManagement>
-			<plugins>
+		<pluginManagement>
+			<plugins>
+				<plugin>
+					<groupId>org.springframework.boot</groupId>
+					<artifactId>spring-boot-maven-plugin</artifactId>
+					<configuration>
+						<layers>
+							<enabled>true</enabled>
+						</layers>
+						<excludes>
+							<exclude>
+								<groupId>org.projectlombok</groupId>
+								<artifactId>lombok</artifactId>
+							</exclude>
+						</excludes>
+					</configuration>
+				</plugin>
 				<plugin>
 					<groupId>org.apache.maven.plugins</groupId>
 					<artifactId>maven-surefire-plugin</artifactId>
@@ -160,7 +76,6 @@
 							<artifactId>checkstyle</artifactId>
 							<version>10.3</version>
 						</dependency>
-
 					</dependencies>
 				</plugin>
 				<plugin>
@@ -274,16 +189,5 @@
 				</plugins>
 			</reporting>
 		</profile>
-		<profile>
-			<id>coverage</id>
-			<build>
-				<plugins>
-					<plugin>
-						<groupId>org.jacoco</groupId>
-						<artifactId>jacoco-maven-plugin</artifactId>
-					</plugin>
-				</plugins>
-			</build>
-		</profile>
 	</profiles>
-</project>
+</project>
\ No newline at end of file
Index: src/main/java/ru/practicum/shareit/user/UserController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/practicum/shareit/user/UserController.java b/server/src/main/java/ru/practicum/shareit/user/UserController.java
rename from src/main/java/ru/practicum/shareit/user/UserController.java
rename to server/src/main/java/ru/practicum/shareit/user/UserController.java
--- a/src/main/java/ru/practicum/shareit/user/UserController.java	(revision 67e65e866b978068413d04956e35ebed1b747536)
+++ b/server/src/main/java/ru/practicum/shareit/user/UserController.java	(date 1728376814910)
@@ -1,8 +1,6 @@
 package ru.practicum.shareit.user;
 
-import jakarta.validation.Valid;
 import lombok.RequiredArgsConstructor;
-import org.springframework.validation.annotation.Validated;
 import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PatchMapping;
@@ -18,7 +16,6 @@
 @RestController
 @RequestMapping("/users")
 @RequiredArgsConstructor
-@Validated
 public class UserController {
 
     private final UserService userService;
@@ -29,12 +26,12 @@
     }
 
     @PostMapping
-    public UserDto createUser(@Valid @RequestBody NewUserRequestDto request) {
+    public UserDto createUser(@RequestBody NewUserRequestDto request) {
         return userService.createUser(request);
     }
 
     @PatchMapping("/{userId}")
-    public UserDto updateUser(@Valid @RequestBody UpdateUserRequestDto request,
+    public UserDto updateUser(@RequestBody UpdateUserRequestDto request,
                               @PathVariable long userId) {
         return userService.updateUser(request, userId);
     }
Index: src/main/java/ru/practicum/shareit/booking/BookingController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/practicum/shareit/booking/BookingController.java b/server/src/main/java/ru/practicum/shareit/booking/BookingController.java
rename from src/main/java/ru/practicum/shareit/booking/BookingController.java
rename to server/src/main/java/ru/practicum/shareit/booking/BookingController.java
--- a/src/main/java/ru/practicum/shareit/booking/BookingController.java	(revision 67e65e866b978068413d04956e35ebed1b747536)
+++ b/server/src/main/java/ru/practicum/shareit/booking/BookingController.java	(date 1728375038168)
@@ -1,7 +1,5 @@
 package ru.practicum.shareit.booking;
 
-import jakarta.validation.Valid;
-import jakarta.validation.constraints.Positive;
 import lombok.RequiredArgsConstructor;
 import org.springframework.validation.annotation.Validated;
 import org.springframework.web.bind.annotation.GetMapping;
@@ -29,20 +27,20 @@
     private final BookingService bookingService;
 
     @PostMapping
-    public BookingDto createBooking(@Valid @RequestBody NewBookingRequest bookingRequest,
+    public BookingDto createBooking(@RequestBody NewBookingRequest bookingRequest,
                                     @RequestHeader(USER_ID_HEADER) long userId) {
         return bookingService.createBooking(bookingRequest, userId);
     }
 
     @PatchMapping("/{bookingId}")
-    public BookingDto approvedBooking(@PathVariable @Positive long bookingId,
+    public BookingDto approvedBooking(@PathVariable long bookingId,
                                       @RequestParam("approved") boolean isApproved,
                                       @RequestHeader(USER_ID_HEADER) long userId) {
         return bookingService.approveBooking(userId, bookingId, isApproved);
     }
 
     @GetMapping("{bookingId}")
-    public BookingDto getBooking(@PathVariable @Positive long bookingId,
+    public BookingDto getBooking(@PathVariable long bookingId,
                                  @RequestHeader(USER_ID_HEADER) long userId) {
         return bookingService.getBooking(userId, bookingId);
     }
Index: src/main/java/ru/practicum/shareit/request/ItemRequestController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/practicum/shareit/request/ItemRequestController.java b/server/src/main/java/ru/practicum/shareit/request/ItemRequestController.java
rename from src/main/java/ru/practicum/shareit/request/ItemRequestController.java
rename to server/src/main/java/ru/practicum/shareit/request/ItemRequestController.java
--- a/src/main/java/ru/practicum/shareit/request/ItemRequestController.java	(revision 67e65e866b978068413d04956e35ebed1b747536)
+++ b/server/src/main/java/ru/practicum/shareit/request/ItemRequestController.java	(date 1728376814898)
@@ -1,9 +1,15 @@
 package ru.practicum.shareit.request;
 
 import lombok.RequiredArgsConstructor;
-import org.springframework.web.bind.annotation.*;
-import ru.practicum.shareit.request.dto.RequestDto;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
 import ru.practicum.shareit.request.dto.NewItemRequestDto;
+import ru.practicum.shareit.request.dto.RequestDto;
 
 import java.util.List;
 
Index: src/main/java/ru/practicum/shareit/booking/BookingServiceImpl.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/practicum/shareit/booking/BookingServiceImpl.java b/server/src/main/java/ru/practicum/shareit/booking/BookingServiceImpl.java
rename from src/main/java/ru/practicum/shareit/booking/BookingServiceImpl.java
rename to server/src/main/java/ru/practicum/shareit/booking/BookingServiceImpl.java
--- a/src/main/java/ru/practicum/shareit/booking/BookingServiceImpl.java	(revision 67e65e866b978068413d04956e35ebed1b747536)
+++ b/server/src/main/java/ru/practicum/shareit/booking/BookingServiceImpl.java	(date 1728373261381)
@@ -1,5 +1,6 @@
 package ru.practicum.shareit.booking;
 
+
 import com.querydsl.core.types.dsl.BooleanExpression;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
Index: src/main/resources/application.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>ISO-8859-1
===================================================================
diff --git a/src/main/resources/application.properties b/server/src/main/resources/application.properties
rename from src/main/resources/application.properties
rename to server/src/main/resources/application.properties
--- a/src/main/resources/application.properties	(revision 67e65e866b978068413d04956e35ebed1b747536)
+++ b/server/src/main/resources/application.properties	(date 1728366690283)
@@ -1,3 +1,4 @@
+server.port=9090
 spring.jpa.hibernate.ddl-auto=none
 spring.jpa.properties.hibernate.format_sql=true
 spring.sql.init.mode=always
Index: src/main/java/ru/practicum/shareit/item/dto/NewCommentRequestDto.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/practicum/shareit/item/dto/NewCommentRequestDto.java b/server/src/main/java/ru/practicum/shareit/item/dto/NewCommentRequestDto.java
rename from src/main/java/ru/practicum/shareit/item/dto/NewCommentRequestDto.java
rename to server/src/main/java/ru/practicum/shareit/item/dto/NewCommentRequestDto.java
--- a/src/main/java/ru/practicum/shareit/item/dto/NewCommentRequestDto.java	(revision 67e65e866b978068413d04956e35ebed1b747536)
+++ b/server/src/main/java/ru/practicum/shareit/item/dto/NewCommentRequestDto.java	(date 1728372310799)
@@ -1,13 +1,9 @@
 package ru.practicum.shareit.item.dto;
 
-import jakarta.validation.constraints.NotBlank;
-import jakarta.validation.constraints.Size;
 import lombok.Data;
 
 @Data
 public class NewCommentRequestDto {
 
-    @NotBlank
-    @Size(max = 2000)
     private String text;
 }
Index: src/main/java/ru/practicum/shareit/request/dto/NewItemRequestDto.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/practicum/shareit/request/dto/NewItemRequestDto.java b/server/src/main/java/ru/practicum/shareit/request/dto/NewItemRequestDto.java
rename from src/main/java/ru/practicum/shareit/request/dto/NewItemRequestDto.java
rename to server/src/main/java/ru/practicum/shareit/request/dto/NewItemRequestDto.java
--- a/src/main/java/ru/practicum/shareit/request/dto/NewItemRequestDto.java	(revision 67e65e866b978068413d04956e35ebed1b747536)
+++ b/server/src/main/java/ru/practicum/shareit/request/dto/NewItemRequestDto.java	(date 1728376814935)
@@ -1,13 +1,9 @@
 package ru.practicum.shareit.request.dto;
 
-import jakarta.validation.constraints.NotBlank;
-import jakarta.validation.constraints.Size;
 import lombok.Data;
 
 @Data
 public class NewItemRequestDto {
 
-    @NotBlank
-    @Size(max = 255)
     private String description;
 }
Index: src/main/java/ru/practicum/shareit/request/mapper/ItemRequestMapper.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/practicum/shareit/request/mapper/ItemRequestMapper.java b/server/src/main/java/ru/practicum/shareit/request/mapper/ItemRequestMapper.java
rename from src/main/java/ru/practicum/shareit/request/mapper/ItemRequestMapper.java
rename to server/src/main/java/ru/practicum/shareit/request/mapper/ItemRequestMapper.java
--- a/src/main/java/ru/practicum/shareit/request/mapper/ItemRequestMapper.java	(revision 67e65e866b978068413d04956e35ebed1b747536)
+++ b/server/src/main/java/ru/practicum/shareit/request/mapper/ItemRequestMapper.java	(date 1728376814938)
@@ -8,14 +8,14 @@
 import ru.practicum.shareit.item.dto.ItemRequestDto;
 import ru.practicum.shareit.item.mapper.ItemMapper;
 import ru.practicum.shareit.item.model.Item;
-import ru.practicum.shareit.request.dto.RequestDto;
 import ru.practicum.shareit.request.dto.NewItemRequestDto;
+import ru.practicum.shareit.request.dto.RequestDto;
 import ru.practicum.shareit.request.model.ItemRequest;
 import ru.practicum.shareit.user.model.User;
 
 @Mapper(componentModel = MappingConstants.ComponentModel.SPRING,
         nullValueMapMappingStrategy = NullValueMappingStrategy.RETURN_DEFAULT,
-uses = ItemMapper.class)
+        uses = ItemMapper.class)
 public interface ItemRequestMapper {
 
     ItemRequest mapToItemRequest(NewItemRequestDto request);
Index: src/main/java/ru/practicum/shareit/request/ItemRequestService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/practicum/shareit/request/ItemRequestService.java b/server/src/main/java/ru/practicum/shareit/request/ItemRequestService.java
rename from src/main/java/ru/practicum/shareit/request/ItemRequestService.java
rename to server/src/main/java/ru/practicum/shareit/request/ItemRequestService.java
--- a/src/main/java/ru/practicum/shareit/request/ItemRequestService.java	(revision 67e65e866b978068413d04956e35ebed1b747536)
+++ b/server/src/main/java/ru/practicum/shareit/request/ItemRequestService.java	(date 1728376814959)
@@ -1,7 +1,7 @@
 package ru.practicum.shareit.request;
 
-import ru.practicum.shareit.request.dto.RequestDto;
 import ru.practicum.shareit.request.dto.NewItemRequestDto;
+import ru.practicum.shareit.request.dto.RequestDto;
 
 import java.util.List;
 
Index: src/main/java/ru/practicum/shareit/request/ItemRequestServiceImpl.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/practicum/shareit/request/ItemRequestServiceImpl.java b/server/src/main/java/ru/practicum/shareit/request/ItemRequestServiceImpl.java
rename from src/main/java/ru/practicum/shareit/request/ItemRequestServiceImpl.java
rename to server/src/main/java/ru/practicum/shareit/request/ItemRequestServiceImpl.java
--- a/src/main/java/ru/practicum/shareit/request/ItemRequestServiceImpl.java	(revision 67e65e866b978068413d04956e35ebed1b747536)
+++ b/server/src/main/java/ru/practicum/shareit/request/ItemRequestServiceImpl.java	(date 1728376814925)
@@ -5,9 +5,8 @@
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 import ru.practicum.shareit.exception.EntityNotFoundException;
-import ru.practicum.shareit.item.mapper.ItemMapper;
-import ru.practicum.shareit.request.dto.RequestDto;
 import ru.practicum.shareit.request.dto.NewItemRequestDto;
+import ru.practicum.shareit.request.dto.RequestDto;
 import ru.practicum.shareit.request.mapper.ItemRequestMapper;
 import ru.practicum.shareit.request.model.ItemRequest;
 import ru.practicum.shareit.user.UserService;
Index: src/main/java/ru/practicum/shareit/user/dto/NewUserRequestDto.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/practicum/shareit/user/dto/NewUserRequestDto.java b/server/src/main/java/ru/practicum/shareit/user/dto/NewUserRequestDto.java
rename from src/main/java/ru/practicum/shareit/user/dto/NewUserRequestDto.java
rename to server/src/main/java/ru/practicum/shareit/user/dto/NewUserRequestDto.java
--- a/src/main/java/ru/practicum/shareit/user/dto/NewUserRequestDto.java	(revision 67e65e866b978068413d04956e35ebed1b747536)
+++ b/server/src/main/java/ru/practicum/shareit/user/dto/NewUserRequestDto.java	(date 1728376814946)
@@ -1,18 +1,11 @@
 package ru.practicum.shareit.user.dto;
 
-import jakarta.validation.constraints.Email;
-import jakarta.validation.constraints.NotBlank;
-import jakarta.validation.constraints.Size;
 import lombok.Data;
 
 @Data
 public class NewUserRequestDto {
 
     private long id;
-    @NotBlank
     private String name;
-    @NotBlank
-    @Email
-    @Size(max = 255)
     private String email;
 }
Index: src/main/java/ru/practicum/shareit/item/dto/UpdateItemRequestDto.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/practicum/shareit/item/dto/UpdateItemRequestDto.java b/server/src/main/java/ru/practicum/shareit/item/dto/UpdateItemRequestDto.java
rename from src/main/java/ru/practicum/shareit/item/dto/UpdateItemRequestDto.java
rename to server/src/main/java/ru/practicum/shareit/item/dto/UpdateItemRequestDto.java
--- a/src/main/java/ru/practicum/shareit/item/dto/UpdateItemRequestDto.java	(revision 67e65e866b978068413d04956e35ebed1b747536)
+++ b/server/src/main/java/ru/practicum/shareit/item/dto/UpdateItemRequestDto.java	(date 1728372310809)
@@ -1,6 +1,5 @@
 package ru.practicum.shareit.item.dto;
 
-import jakarta.validation.constraints.Size;
 import lombok.Data;
 
 
@@ -8,9 +7,7 @@
 public class UpdateItemRequestDto {
 
     private long id;
-    @Size(max = 255)
     private String name;
-    @Size(max = 2000)
     private String description;
     private Boolean available;
 }
Index: src/main/java/ru/practicum/shareit/item/dto/NewItemRequestDto.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/practicum/shareit/item/dto/NewItemRequestDto.java b/server/src/main/java/ru/practicum/shareit/item/dto/NewItemRequestDto.java
rename from src/main/java/ru/practicum/shareit/item/dto/NewItemRequestDto.java
rename to server/src/main/java/ru/practicum/shareit/item/dto/NewItemRequestDto.java
--- a/src/main/java/ru/practicum/shareit/item/dto/NewItemRequestDto.java	(revision 67e65e866b978068413d04956e35ebed1b747536)
+++ b/server/src/main/java/ru/practicum/shareit/item/dto/NewItemRequestDto.java	(date 1728372310787)
@@ -1,22 +1,14 @@
 package ru.practicum.shareit.item.dto;
 
 
-import jakarta.validation.constraints.NotBlank;
-import jakarta.validation.constraints.NotNull;
-import jakarta.validation.constraints.Size;
 import lombok.Data;
 
 @Data
 public class NewItemRequestDto {
 
     private long id;
-    @NotBlank
-    @Size(max = 255)
     private String name;
-    @NotBlank
-    @Size(max = 2000)
     private String description;
-    @NotNull
     private Boolean available;
     private Long requestId;
 }
Index: src/main/java/ru/practicum/shareit/user/dto/UpdateUserRequestDto.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/practicum/shareit/user/dto/UpdateUserRequestDto.java b/server/src/main/java/ru/practicum/shareit/user/dto/UpdateUserRequestDto.java
rename from src/main/java/ru/practicum/shareit/user/dto/UpdateUserRequestDto.java
rename to server/src/main/java/ru/practicum/shareit/user/dto/UpdateUserRequestDto.java
--- a/src/main/java/ru/practicum/shareit/user/dto/UpdateUserRequestDto.java	(revision 67e65e866b978068413d04956e35ebed1b747536)
+++ b/server/src/main/java/ru/practicum/shareit/user/dto/UpdateUserRequestDto.java	(date 1728376814921)
@@ -1,6 +1,5 @@
 package ru.practicum.shareit.user.dto;
 
-import jakarta.validation.constraints.Email;
 import lombok.Data;
 
 @Data
@@ -8,6 +7,5 @@
 
     private long id;
     private String name;
-    @Email
     private String email;
 }
diff --git a/src/main/java/ru/practicum/shareit/item/dto/ItemDto.java b/server/src/main/java/ru/practicum/shareit/item/dto/ItemDto.java
rename from src/main/java/ru/practicum/shareit/item/dto/ItemDto.java
rename to server/src/main/java/ru/practicum/shareit/item/dto/ItemDto.java
diff --git a/src/main/java/ru/practicum/shareit/item/model/Item.java b/server/src/main/java/ru/practicum/shareit/item/model/Item.java
rename from src/main/java/ru/practicum/shareit/item/model/Item.java
rename to server/src/main/java/ru/practicum/shareit/item/model/Item.java
diff --git a/src/main/java/ru/practicum/shareit/booking/dto/BookingDto.java b/server/src/main/java/ru/practicum/shareit/booking/dto/BookingDto.java
rename from src/main/java/ru/practicum/shareit/booking/dto/BookingDto.java
rename to server/src/main/java/ru/practicum/shareit/booking/dto/BookingDto.java
diff --git a/src/main/java/ru/practicum/shareit/request/dto/RequestDto.java b/server/src/main/java/ru/practicum/shareit/request/dto/RequestDto.java
rename from src/main/java/ru/practicum/shareit/request/dto/RequestDto.java
rename to server/src/main/java/ru/practicum/shareit/request/dto/RequestDto.java
diff --git a/src/main/java/ru/practicum/shareit/booking/SearchBookingStates.java b/server/src/main/java/ru/practicum/shareit/booking/SearchBookingStates.java
rename from src/main/java/ru/practicum/shareit/booking/SearchBookingStates.java
rename to server/src/main/java/ru/practicum/shareit/booking/SearchBookingStates.java
diff --git a/src/main/java/ru/practicum/shareit/booking/model/Booking.java b/server/src/main/java/ru/practicum/shareit/booking/model/Booking.java
rename from src/main/java/ru/practicum/shareit/booking/model/Booking.java
rename to server/src/main/java/ru/practicum/shareit/booking/model/Booking.java
diff --git a/src/main/java/ru/practicum/shareit/booking/model/BookingStatus.java b/server/src/main/java/ru/practicum/shareit/booking/model/BookingStatus.java
rename from src/main/java/ru/practicum/shareit/booking/model/BookingStatus.java
rename to server/src/main/java/ru/practicum/shareit/booking/model/BookingStatus.java
diff --git a/src/test/java/ru/practicum/shareit/ShareItTests.java b/server/src/test/java/ru/practicum/shareit/ShareItTests.java
rename from src/test/java/ru/practicum/shareit/ShareItTests.java
rename to server/src/test/java/ru/practicum/shareit/ShareItTests.java
diff --git a/src/main/java/ru/practicum/shareit/item/CommentRepository.java b/server/src/main/java/ru/practicum/shareit/item/CommentRepository.java
rename from src/main/java/ru/practicum/shareit/item/CommentRepository.java
rename to server/src/main/java/ru/practicum/shareit/item/CommentRepository.java
diff --git a/src/main/java/ru/practicum/shareit/booking/BookingRepository.java b/server/src/main/java/ru/practicum/shareit/booking/BookingRepository.java
rename from src/main/java/ru/practicum/shareit/booking/BookingRepository.java
rename to server/src/main/java/ru/practicum/shareit/booking/BookingRepository.java
diff --git a/src/main/java/ru/practicum/shareit/booking/BookingService.java b/server/src/main/java/ru/practicum/shareit/booking/BookingService.java
rename from src/main/java/ru/practicum/shareit/booking/BookingService.java
rename to server/src/main/java/ru/practicum/shareit/booking/BookingService.java
diff --git a/src/main/java/ru/practicum/shareit/booking/mapper/BookingMapper.java b/server/src/main/java/ru/practicum/shareit/booking/mapper/BookingMapper.java
rename from src/main/java/ru/practicum/shareit/booking/mapper/BookingMapper.java
rename to server/src/main/java/ru/practicum/shareit/booking/mapper/BookingMapper.java
diff --git a/src/main/java/ru/practicum/shareit/item/model/Comment.java b/server/src/main/java/ru/practicum/shareit/item/model/Comment.java
rename from src/main/java/ru/practicum/shareit/item/model/Comment.java
rename to server/src/main/java/ru/practicum/shareit/item/model/Comment.java
diff --git a/src/main/java/ru/practicum/shareit/exception/ItemIsNotAvailable.java b/server/src/main/java/ru/practicum/shareit/exception/ItemIsNotAvailable.java
rename from src/main/java/ru/practicum/shareit/exception/ItemIsNotAvailable.java
rename to server/src/main/java/ru/practicum/shareit/exception/ItemIsNotAvailable.java
diff --git a/src/main/java/ru/practicum/shareit/exception/NotEnoughRightsToChangeData.java b/server/src/main/java/ru/practicum/shareit/exception/NotEnoughRightsToChangeData.java
rename from src/main/java/ru/practicum/shareit/exception/NotEnoughRightsToChangeData.java
rename to server/src/main/java/ru/practicum/shareit/exception/NotEnoughRightsToChangeData.java
diff --git a/src/main/java/ru/practicum/shareit/exception/EmailAlreadyExist.java b/server/src/main/java/ru/practicum/shareit/exception/EmailAlreadyExist.java
rename from src/main/java/ru/practicum/shareit/exception/EmailAlreadyExist.java
rename to server/src/main/java/ru/practicum/shareit/exception/EmailAlreadyExist.java
diff --git a/src/main/resources/schema.sql b/server/src/main/resources/schema.sql
rename from src/main/resources/schema.sql
rename to server/src/main/resources/schema.sql
diff --git a/src/main/java/ru/practicum/shareit/item/dto/CommentDto.java b/server/src/main/java/ru/practicum/shareit/item/dto/CommentDto.java
rename from src/main/java/ru/practicum/shareit/item/dto/CommentDto.java
rename to server/src/main/java/ru/practicum/shareit/item/dto/CommentDto.java
diff --git a/src/main/java/ru/practicum/shareit/item/dto/ItemOwnerDto.java b/server/src/main/java/ru/practicum/shareit/item/dto/ItemOwnerDto.java
rename from src/main/java/ru/practicum/shareit/item/dto/ItemOwnerDto.java
rename to server/src/main/java/ru/practicum/shareit/item/dto/ItemOwnerDto.java
diff --git a/src/main/java/ru/practicum/shareit/request/model/ItemRequest.java b/server/src/main/java/ru/practicum/shareit/request/model/ItemRequest.java
rename from src/main/java/ru/practicum/shareit/request/model/ItemRequest.java
rename to server/src/main/java/ru/practicum/shareit/request/model/ItemRequest.java
diff --git a/src/main/java/ru/practicum/shareit/item/mapper/CommentMapper.java b/server/src/main/java/ru/practicum/shareit/item/mapper/CommentMapper.java
rename from src/main/java/ru/practicum/shareit/item/mapper/CommentMapper.java
rename to server/src/main/java/ru/practicum/shareit/item/mapper/CommentMapper.java
diff --git a/src/main/java/ru/practicum/shareit/request/ItemRequestRepository.java b/server/src/main/java/ru/practicum/shareit/request/ItemRequestRepository.java
rename from src/main/java/ru/practicum/shareit/request/ItemRequestRepository.java
rename to server/src/main/java/ru/practicum/shareit/request/ItemRequestRepository.java
diff --git a/src/test/java/ru/practicum/shareit/UserServiceImplTest.java b/server/src/test/java/ru/practicum/shareit/UserServiceImplTest.java
rename from src/test/java/ru/practicum/shareit/UserServiceImplTest.java
rename to server/src/test/java/ru/practicum/shareit/UserServiceImplTest.java
diff --git a/src/main/java/ru/practicum/shareit/exception/EntityNotFoundException.java b/server/src/main/java/ru/practicum/shareit/exception/EntityNotFoundException.java
rename from src/main/java/ru/practicum/shareit/exception/EntityNotFoundException.java
rename to server/src/main/java/ru/practicum/shareit/exception/EntityNotFoundException.java
diff --git a/src/main/java/ru/practicum/shareit/exception/ErrorHandler.java b/server/src/main/java/ru/practicum/shareit/exception/ErrorHandler.java
rename from src/main/java/ru/practicum/shareit/exception/ErrorHandler.java
rename to server/src/main/java/ru/practicum/shareit/exception/ErrorHandler.java
diff --git a/src/main/java/ru/practicum/shareit/exception/ErrorResponse.java b/server/src/main/java/ru/practicum/shareit/exception/ErrorResponse.java
rename from src/main/java/ru/practicum/shareit/exception/ErrorResponse.java
rename to server/src/main/java/ru/practicum/shareit/exception/ErrorResponse.java
diff --git a/src/main/java/ru/practicum/shareit/exception/IncorrectArgumentException.java b/server/src/main/java/ru/practicum/shareit/exception/IncorrectArgumentException.java
rename from src/main/java/ru/practicum/shareit/exception/IncorrectArgumentException.java
rename to server/src/main/java/ru/practicum/shareit/exception/IncorrectArgumentException.java
diff --git a/src/main/java/ru/practicum/shareit/item/dto/ItemRequestDto.java b/server/src/main/java/ru/practicum/shareit/item/dto/ItemRequestDto.java
rename from src/main/java/ru/practicum/shareit/item/dto/ItemRequestDto.java
rename to server/src/main/java/ru/practicum/shareit/item/dto/ItemRequestDto.java
diff --git a/src/main/java/ru/practicum/shareit/item/ItemServiceImpl.java b/server/src/main/java/ru/practicum/shareit/item/ItemServiceImpl.java
rename from src/main/java/ru/practicum/shareit/item/ItemServiceImpl.java
rename to server/src/main/java/ru/practicum/shareit/item/ItemServiceImpl.java
diff --git a/src/main/java/ru/practicum/shareit/item/ItemService.java b/server/src/main/java/ru/practicum/shareit/item/ItemService.java
rename from src/main/java/ru/practicum/shareit/item/ItemService.java
rename to server/src/main/java/ru/practicum/shareit/item/ItemService.java
diff --git a/src/main/java/ru/practicum/shareit/user/UserService.java b/server/src/main/java/ru/practicum/shareit/user/UserService.java
rename from src/main/java/ru/practicum/shareit/user/UserService.java
rename to server/src/main/java/ru/practicum/shareit/user/UserService.java
diff --git a/src/main/java/ru/practicum/shareit/user/UserRepository.java b/server/src/main/java/ru/practicum/shareit/user/UserRepository.java
rename from src/main/java/ru/practicum/shareit/user/UserRepository.java
rename to server/src/main/java/ru/practicum/shareit/user/UserRepository.java
diff --git a/src/main/java/ru/practicum/shareit/item/ItemRepository.java b/server/src/main/java/ru/practicum/shareit/item/ItemRepository.java
rename from src/main/java/ru/practicum/shareit/item/ItemRepository.java
rename to server/src/main/java/ru/practicum/shareit/item/ItemRepository.java
diff --git a/src/main/java/ru/practicum/shareit/item/mapper/ItemMapper.java b/server/src/main/java/ru/practicum/shareit/item/mapper/ItemMapper.java
rename from src/main/java/ru/practicum/shareit/item/mapper/ItemMapper.java
rename to server/src/main/java/ru/practicum/shareit/item/mapper/ItemMapper.java
diff --git a/src/main/java/ru/practicum/shareit/user/model/User.java b/server/src/main/java/ru/practicum/shareit/user/model/User.java
rename from src/main/java/ru/practicum/shareit/user/model/User.java
rename to server/src/main/java/ru/practicum/shareit/user/model/User.java
diff --git a/src/main/java/ru/practicum/shareit/user/UserServiceImpl.java b/server/src/main/java/ru/practicum/shareit/user/UserServiceImpl.java
rename from src/main/java/ru/practicum/shareit/user/UserServiceImpl.java
rename to server/src/main/java/ru/practicum/shareit/user/UserServiceImpl.java
diff --git a/src/main/java/ru/practicum/shareit/user/mapper/UserMapper.java b/server/src/main/java/ru/practicum/shareit/user/mapper/UserMapper.java
rename from src/main/java/ru/practicum/shareit/user/mapper/UserMapper.java
rename to server/src/main/java/ru/practicum/shareit/user/mapper/UserMapper.java
diff --git a/src/main/java/ru/practicum/shareit/util/HeaderConstant.java b/server/src/main/java/ru/practicum/shareit/util/HeaderConstant.java
rename from src/main/java/ru/practicum/shareit/util/HeaderConstant.java
rename to server/src/main/java/ru/practicum/shareit/util/HeaderConstant.java
diff --git a/src/main/java/ru/practicum/shareit/user/dto/UserDto.java b/server/src/main/java/ru/practicum/shareit/user/dto/UserDto.java
rename from src/main/java/ru/practicum/shareit/user/dto/UserDto.java
rename to server/src/main/java/ru/practicum/shareit/user/dto/UserDto.java
